### PR TITLE
Remove ValidateFromChainState

### PR DESCRIPTION
**Pull request by AndreiEres**
# Description

This PR removes the `CandidateValidationMessage::ValidateFromChainState`, which was previously used by backing, but is no longer relevant since initial async backing implementation https://github.com/paritytech/polkadot/pull/5557. 

Fixes https://github.com/paritytech/polkadot-sdk/issues/5643

## Integration

This change should not affect downstream projects since `ValidateFromChainState` was already unused.

## Review Notes

- Removed all occurrences of `ValidateFromChainState`.
- Moved utility functions, previously used in candidate validation tests and malus, exclusively to candidate validation tests as they are no longer used in malus.
- Deleted the `polkadot_parachain_candidate_validation_validate_from_chain_state` metric from Prometheus.
- Removed `Spawner` from `ReplaceValidationResult` in malus’ interceptors.
- `fake_validation_error` was only used for `ValidateFromChainState` handling, while other cases directly used `InvalidCandidate::InvalidOutputs`. It has been replaced with `fake_validation_error`, with a fallback to `InvalidCandidate::InvalidOutputs`.
- Updated overseer’s minimal example to replace `ValidateFromChainState` with `ValidateFromExhaustive`.

### CODE REVIEW

File: polkadot/node/core/candidate-validation/src/tests.rs [hunk 1]
```diff
@@ -34,6 +35,58 @@ use sp_core::{sr25519::Public, testing::TaskExecutor};
...
 use sp_keystore::{testing::MemoryKeystore, Keystore};
 
+#[derive(Debug)]
+enum AssumptionCheckOutcome {
+	Matches(PersistedValidationData, ValidationCode),
+	DoesNotMatch,
+	BadRequest,
+}
+
+async fn check_assumption_validation_data<Sender>(
```
Comment (alexggh) [inline polkadot/node/core/candidate-validation/src/tests.rs:45]: Why do we need this here ?

**Review by alexggh — APPROVED** [review]

File: polkadot/node/core/candidate-validation/src/tests.rs (hunk 1 above)
Comment (AndreiEres) [reply polkadot/node/core/candidate-validation/src/tests.rs:45]: It's used bellow in tests.

Comment (AndreiEres) [reply polkadot/node/core/candidate-validation/src/tests.rs:45]: From Review Notes in the description
> Moved utility functions, previously used in candidate validation tests and malus, exclusively to candidate validation tests as they are no longer used in malus.

**Review by s0me0ne-unkn0wn — APPROVED** [review]
I love where it's going, first we removed jaeger, then a subsystem message variant, now I eagerly await when we start to remove whole subsystems :)

**Review by sandreim — APPROVED** [review]
Nice! LGTM!

**Pull request merged**